lib/GHCup/Types.hs

```diff
@@ -394,6 +394,13 @@ data NewURLSource = NewGHCupURL
 
 instance NFData NewURLSource
 
+-- | Alias for ease of URLSource selection
+data ChannelAlias = MainChannel
+                  | CrossChannel
+                  | PrereleasesChannel
+                  | VanillaChannel
```

Comment:
This is now a bit confusing, because the config supports the value `GHCupURL` and `StackSetupURL`, but none of the other channels.

https://github.com/haskell/ghcup-hs/blob/a32a5597300b777b488e1649b1a662dc5cce150f/data/config.yaml#L54-L92

https://github.com/haskell/ghcup-hs/blob/a32a5597300b777b488e1649b1a662dc5cce150f/lib/GHCup/Types.hs#L388-L393

And the json instances are finicky.

Reply:
It creates inconsistency, because `NewURLSource` is valid in both the config and in the cli: `ghcup -s GHCupURL tui`.

Now we have another set of aliases that don't work in the config but in the cli. I have tried to keep both of them in sync as much as possible.

Reply:
Now the alias is part of `URLSource` & `NewURLSource`

Reply:
Correction alias is only part of `NewURLSource`